app/controllers/hostgroups_controller.rb

```diff
@@ -40,10 +40,17 @@ def nest
   # Clone the hostgroup
   def clone
     new = @hostgroup.clone
+    new.name = nil
     load_vars_for_ajax
+    # Clone all its facets
+    @hostgroup.facet_definitions.map(&:name).each do |facet_name|
```

Comment:
I think you don't need to do the legwork manually, you can use the `SelectiveClone` to do the work for you, similar to the host object: https://github.com/theforeman/foreman/blob/c31702337015ecdf76aaa3053924b57007b8867f/app/models/concerns/facets/managed_host_extensions.rb#L10